spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestUpdate.java

```diff
@@ -687,6 +696,13 @@ public synchronized void testUpdateWithSnapshotIsolation()
       throws InterruptedException, ExecutionException {
     // cannot run tests with concurrency for Hadoop tables without atomic renames
     Assume.assumeFalse(catalogName.equalsIgnoreCase("testhadoop"));
+    // if caching is off, the table is eagerly refreshed during runtime filtering
+    // this can cause a validation exception as concurrent changes would be visible
```

Comment:
Should we extract this code to a method in the `SparkRowLevelOperationsTestBase`? If so, we can reduce the duplicated code and comment.

Reply:
I added a common method to check whether caching is enabled, but I feel the comment must be in the test itself for better visibility.